src/mongo/db/commands/resize_oplog.cpp

```diff
@@ -0,0 +1,111 @@
+*    must comply with the GNU Affero General Public License in all respects for
+*    all of the code used other than as permitted herein. If you modify file(s)
+*    with this exception, you may extend this exception to your version of the
+*    file(s), but you are not obligated to do so. If you do not wish to do so,
+*    delete this exception statement from your version. If you delete this
+*    exception statement from all source files in the program, then also delete
+*    it in the license file.
+*/
+
+#define MONGO_LOG_DEFAULT_COMPONENT ::mongo::logger::LogComponent::kCommand
+
+#include "mongo/platform/basic.h"
+
+#include <string>
+#include <vector>
+
+#include "mongo/db/auth/authorization_manager.h"
+#include "mongo/db/auth/authorization_manager_global.h"
+#include "mongo/db/auth/authorization_session.h"
+#include "mongo/db/catalog/apply_ops.h"
+#include "mongo/db/client.h"
+#include "mongo/db/commands.h"
+#include "mongo/db/commands/dbhash.h"
+#include "mongo/db/concurrency/write_conflict_exception.h"
+#include "mongo/db/db_raii.h"
+#include "mongo/db/dbdirectclient.h"
+#include "mongo/db/jsobj.h"
+#include "mongo/db/matcher/matcher.h"
+#include "mongo/db/operation_context.h"
+#include "mongo/db/repl/oplog.h"
+#include "mongo/db/repl/repl_client_info.h"
+#include "mongo/db/repl/replication_coordinator_global.h"
+#include "mongo/db/catalog/collection.h"
+#include "mongo/db/catalog/collection_catalog_entry.h"
+#include "mongo/db/service_context.h"
+#include "mongo/util/log.h"
+#include "mongo/util/scopeguard.h"
+
+namespace mongo {
+
+using std::string;
+using std::stringstream;
+
+class CmdResizeOplog : public Command {
+public:
+    virtual bool slaveOk() const {
+        return true;
+    }
+    virtual bool supportsWriteConcern(const BSONObj& cmd) const override {
+        return false;
+    }
+    virtual void help(stringstream& help) const {
+        help << "resize oplog size, only support wiredTiger";
+    }
+    CmdResizeOplog() : Command("resizeOplog") {}
+    bool run(OperationContext* txn,
+            const string& dbname,
+            BSONObj& jsobj,
+            int,
+            string& errmsg,
+            BSONObjBuilder& result) {
+        StringData dbName("local");
+        ScopedTransaction transaction(txn, MODE_IX);
+        AutoGetDb autoDb(txn, dbName, MODE_X);
+        Database* const db = autoDb.getDb();
+        Collection* coll = db ? db->getCollection("local.oplog.rs") : nullptr;
+        if (!coll) {
+            return appendCommandStatus(result, Status(ErrorCodes::NamespaceNotFound, "ns does not exist"));
+        }
+        if (!coll->isCapped()) {
+            return appendCommandStatus(result, Status(ErrorCodes::InternalError, "ns does not exist"));
+        }
+        if (!jsobj["size"].isNumber()) {
+            return appendCommandStatus(result, Status(ErrorCodes::InvalidOptions, "invalid size field, size should be a number"));
+        }
+
+        long long size = jsobj["size"].numberLong();
```

Comment:
The minimum size is 990 MB (`990LL * 1024 * 1024` bytes, see `oplog.cpp`), so that should be checked. Would be even better to move that constant to `oplog.h`.